contracts/Succinct_SpokePool.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./SpokePool.sol";
+import "./interfaces/WETH9Interface.sol";
+import "./external/SuccinctInterfaces.sol";
+
+contract Succinct_SpokePool is SpokePool, ITelepathyHandler {
+    // Address of the succinct contract.
+    address public succinctTargetAmb;
+
+    // Warning: this variable should _never_ be touched outside of this contract. It is intentionally set to be
+    // private. Leaving it set to true can permanently disable admin calls.
+    bool private adminCallValidated;
+
+    uint16 public hubChainId;
```

Comment:
nit: can we make this a constant = 1 and temporarily override when deploying to goerli? Not a big deal but save some gas

Reply:
I don't think we can do that (unless we make it immutable) because it would be stored in the implementation contract and not the proxy, right?